### ISSUE TITLE
Unnable to invoke function called 'then'

### ISSUE THREAD

**Issue by kasperlanger** [puffnfresh/roy#61]
Is there a way to invoke methods called `then` like in the following example

```
($.when "foo").then(\x ->
  console.log x
)
```

I was hoping it would compile to something like

```
$.when("foo").then(function(x){console.log(x);});
```

but I get the following error

```
Error: Parse error on line 2: Unexpected 'THEN'
```

**Comment by puffnfresh** [comment]
Fixed in 9fce2052e3badd8747e9df788af38cf8fc3333f7

**Comment by kasperlanger** [comment]
Awesome thanks!
